### PR TITLE
Add high level feature overview documentation

### PR DESCRIPTION
**Pull request by quba42**
fixes #7318
https://pulp.plan.io/issues/7318

### CODE REVIEW

File: docs/feature_overview.rst [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+Repository Synchronization
+--------------------------------------------------------------------------------
+
+Synchronizing upstream repositories is one of two ways to obtain content for the ``pulp_deb`` plugin.
+(The other one being :ref:`package uploads <package_uploads>`.)
+
+The aim is for the plugin to be able to synchronize (and usably publish) arbitrary (valid) APT repositories.
+Most of the testing is based on the structure of the official Debian and Ubuntu repositories.
+As of this writing all testing happens with regular binary package repositories only.
```
Comment (mdellweg) [inline docs/feature_overview.rst:24]: Isn't there at least a udeb and maybe one installer_file in the fixtures?

Comment (mdellweg) [reply docs/feature_overview.rst:24]: Also what is the point of talking about tests in a "feature overview"?

Comment (quba42) [reply docs/feature_overview.rst:24]: I guess I was not very clear on supported package types.
Perhaps I should make that a completely separate heading.
The point is to give some indication what can really be expected to work for sure.

We do have a separate model for udebs (`InstallerPackage` I think).
However, if I have understood things correctly (and please correct me if I am wrong), those are only usable in conjunction with the verbatim publisher right now?

I don't think any of us have ever tried to upload source packages? (If they work, then by accident?)

Linux Mint repos (for example) should work, but I don't think I have ever tested one.

That being said, I will rework this paragraph (once I have cleared up some of these points for myself).

Comment (mdellweg) [reply docs/feature_overview.rst:24]: Publishing udebs without the 'other' installer files doesn't bear any value, so it is at least only useful in conjunction with the verbatim publisher atm.

**Comment by quba42** [comment]
In the interest of reviewability I have split out https://github.com/pulp/pulp_deb/pull/202 into a separate "spin off PR".
I still think both are required for a complete "feature overview" documentation.

**Comment by quba42** [comment]
Added many suggestion by @maximiliankolb

**Pull request merged**